task02/queue.cpp

```diff
@@ -0,0 +1,75 @@
+#include "queue.h"
+#include <iostream>
+#include <stdlib.h>
+
+ListQueue :: ListQueue (){
+    m_head = NULL;
+}
+
+void ListQueue :: enqueue (data &a){
+    if (Node* node = new Node(a)){
+        node->m_next = m_head;
+        m_head = node;
+    }
+}
+
+data ListQueue :: dequeue () {
+    if (m_head) {
+        if(this->getLength()==1)
+        return m_head->m_t;
+        Node* newHead = m_head;
+        int i = 0;
+        while(i<(this->getLength()-2)){

```

Comment:
Very insufficient solution... dequeue works with O(n^2) computational complexity (every cycle getLength() is called - O(n)).
Try to store another pointer to tail, it should speed up algorithm (you will need to revert list to another direction).